### ISSUE TITLE
SMRT produces all NaN

### ISSUE THREAD

**Issue by jasonw247** [tgsmith61591/smrt#10]
#Sample code

```python
import numpy as np
from sklearn import datasets
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import train_test_split
from smrt.balance import smrt_balance
import pandas as pd

seed = 42

def find_null_rows(df):
    idx = df.index[df.isnull().all(1)]
    nans = df.iloc[idx]
    return nans


# Generate the dataset
X, y = datasets.make_classification(n_classes=2, class_sep=2,
                                    weights=[.95, .05], n_informative=50,
                                    n_redundant=10, flip_y=0, n_features=100,
                                    n_clusters_per_class=4, n_samples=50000,
                                    random_state=seed)

X_train, X_test, y_train, y_test = train_test_split(X,y, test_size=.3)

train_data = X_train
label_data = y_train
X_bal, y_bal = smrt_balance(train_data,label_data, shuffle=False, balance_ratio = .1,
                            n_hidden=500, n_latent_factors=10 ,activation_function='relu',
                            learning_function='adam',clip=True,layer_type='xavier',
                           bias_strategy='ones',early_stopping=True)

balanced_ex = pd.DataFrame(X_bal)

#count the number of null rows
null_vals = find_null_rows(balanced_ex)
print(len(null_vals))

#view the difference between x_train and balanced datasets
#so far, have only gotten entirely null rows to be generated
print(len(X_train))
print(len(X_bal))
```

**Comment by tgsmith61591** [comment]
I was able to produce no `NaN` values if I used `activation_function='sigmoid'`. Also, I made sure to `stratify` on `y` for the `train_test_split` (but I don't think that was the issue). Since all of the activation functions are using the tensorflow API, I think if there is any issue at all, it's on their side. This is definitely a vanishing gradient situation.

**Comment by octavflorescu** [comment]
Hi, i have also encountered SMRT NAN issue.
I have also tested the VAE on the same input, and the cost is NAN.

Tested on a subset of CIFAR10
https://github.com/octavflorescu/sharing/blob/master/SMRT.ipynb

(
also there is a bug on smrt.py line 207:
fix:         X_sub = X_copy[y_transform == transformed_label, :]
)

**Comment by tgsmith61591** [comment]
PRs always welcome :-)

**Comment by hasibzunair** [comment]
@tgsmith61591 trying to train the VAE using the `VariationalAutoEncoder` class. My batch of image are of shape `(3389, 256, 256, 3)` which are reshaped to `(3389, 196608)` for training. I am getting `cost=nan` everytime. Below is the training code.

```
from smrt.autoencode import VariationalAutoEncoder

v_encoder = VariationalAutoEncoder(n_epochs=400, n_hidden=300, n_latent_factors=15, 
                                   learning_rate=0.075, batch_size=128, display_step=1, 
                                   activation_function='sigmoid', verbose=2, l2_penalty=None,
                                   random_state=seed, early_stopping=True, dropout=0.4,
                                   learning_function='sgd', clip=False)

v_encoder.fit(hem_images)
```

I tried the example code you provided an it works fine. Are there any limitations for RGB images?

**Comment by tgsmith61591** [comment]
It's a 2-yo repository that is mostly dormant at this point. If you can track it down, we're more than happy to look at a PR but both myself and @jasonw247 have been sucked into other projects at this point.

**Comment by hasibzunair** [comment]
Alright.
